### DISCUSSION TITLE
How to obtain unsaved changed value in the muiTableBodyCellEditTextFieldProps

### DISCUSSION THREAD

**Discussion by AdityaBhadange** [KevinVandy/material-react-table#290] [Q&A]
Hello, I'm working on the inline edit row feature. Now I have the requirement to validate my edit TextField component with respect to the other fields. I can get the rowData (i.e. old data) from **muiTableBodyCellEditTextFieldProps**. Still, I want the rowData that I have changed but hasn't been saved yet to validate my input with the other cell's updated data on the go.

Last time I was able to do this with mbrn/material-table by mutating the table data cell value directly onChange, so that on the validate field I could get all the updated cell values. But I think that was the wrong approach, so I'm asking here for a cleaner approach.

Hope this clarifies.

![Screenshot from 2022-12-29 21-49-32](https://user-images.githubusercontent.com/77658603/209981363-ee94fc83-9686-4610-ae18-ad628e3ed863.png)

**Comment by KevinVandy** [comment]
Have you seen the validation example code in the Editing CRUD example on the docs site?

**Reply by AdityaBhadange** [reply]
Yes I have checked the docs. However my query is that, suppose I'm editing and putting 7 integer value in the 1st textfield that's type of number, then go to the second field and put 6. Now I want the 2nd field to throw the error because 2nd field number is less than 1st field, and for that I will need the updated row values in each row before saving anything (It's a very specific use case, that most won't need, but here I'm 😅.

Currently I can get the rowdata, but unless I save it my rowdata doesn't gets updated. So I imagining something more like soft update of rowdata as I only type but doesn't save.

Otherwise I would need to manually save the onChange data on each textfield with state variable & then validate..!

**Reply by KevinVandy** [reply]
Noting is stopping you from hooking validation to an `onChange` in  `muiTableBodyCellEditTextFieldProps`. Its even one of the examples in the [Even Listeners Guide](https://www.material-react-table.com/docs/guides/table-event-listeners#add-an-onchange-to-an-edit-textfield)
